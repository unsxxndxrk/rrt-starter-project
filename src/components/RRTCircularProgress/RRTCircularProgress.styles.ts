import { makeStyles } from "@material-ui/core/styles";

const useRRTCircularProgressStyles = makeStyles({
  root: {
    display: "flex",
    width: "100%",
    justifyContent: "center",
    padding: "100px 0",
  },
});

export { useRRTCircularProgressStyles };
