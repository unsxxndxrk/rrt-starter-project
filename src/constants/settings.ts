export const APP_DEFAULT_LANGUAGE: string = "en";
export const API_ENDPOINT: string =
  process.env.REACT_APP_API_ENDPOINT || window.location.origin;
