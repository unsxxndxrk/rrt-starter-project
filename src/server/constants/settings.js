const ORIGIN = "http://localhost";
const PORT = "8000";
const USERS_PATH = "./data/users.json";
const TODOS_PATH = "./data/todos.json";

exports.ORIGIN = ORIGIN;
exports.PORT = PORT;
exports.USERS_PATH = USERS_PATH;
exports.TODOS_PATH = TODOS_PATH;
