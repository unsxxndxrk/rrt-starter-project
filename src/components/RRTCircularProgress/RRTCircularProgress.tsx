import { useRRTCircularProgressStyles } from "./RRTCircularProgress.styles";
import cx from "classnames";
import { CircularProgress } from "@material-ui/core";

export type RRTCircularProgressProps = {
  className?: string;
  keepHeight?: string;
  color?: "primary" | "secondary" | "inherit";
};

function RRTCircularProgress(props: RRTCircularProgressProps) {
  const { className, color, keepHeight } = props;
  const classes = useRRTCircularProgressStyles();
  const rootClasses = cx(classes.root, className);

  return (
    <div
      className={rootClasses}
      style={{
        height: keepHeight,
      }}
    >
      <CircularProgress color={color || "primary"} />
    </div>
  );
}

export default RRTCircularProgress;
