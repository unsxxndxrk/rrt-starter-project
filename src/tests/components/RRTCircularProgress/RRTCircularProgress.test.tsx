import { render } from "tests/settings/render";
import RRTCircularProgress from "components/RRTCircularProgress/RRTCircularProgress";

const classNameMock: string = "class_name_mock";
const keepHeightMock: string = "420px";

describe("<RRTCircularProgress />", () => {
  test("should render RRTCircularProgress component", () => {
    const wrapper = render(<RRTCircularProgress />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  test("should render RRTCircularProgress component with className prop", () => {
    const wrapper = render(<RRTCircularProgress className={classNameMock} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(wrapper.container.getElementsByClassName(classNameMock).length).toBe(
      1,
    );
  });

  test("should render RRTCircularProgress component with keepHeight prop", () => {
    const wrapper = render(<RRTCircularProgress keepHeight={keepHeightMock} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(
      wrapper.container
        .querySelector(".makeStyles-root")
        ?.getAttribute("style"),
    ).toBe(`height: ${keepHeightMock};`);
  });

  test("should render RRTCircularProgress component with color=primary prop", () => {
    const wrapper = render(<RRTCircularProgress color="primary" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(
      wrapper.container.getElementsByClassName(
        "MuiCircularProgress-colorPrimary",
      ).length,
    ).toBe(1);
  });

  test("should render RRTCircularProgress component with color=secondary prop", () => {
    const wrapper = render(<RRTCircularProgress color="secondary" />);
    expect(wrapper.asFragment()).toMatchSnapshot();
    expect(
      wrapper.container.getElementsByClassName(
        "MuiCircularProgress-colorSecondary",
      ).length,
    ).toBe(1);
  });
});
