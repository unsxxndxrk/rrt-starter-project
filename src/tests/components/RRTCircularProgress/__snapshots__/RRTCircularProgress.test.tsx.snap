// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RRTCircularProgress /> should render RRTCircularProgress component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RRTCircularProgress /> should render RRTCircularProgress component with className prop 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root class_name_mock"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RRTCircularProgress /> should render RRTCircularProgress component with color=primary prop 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RRTCircularProgress /> should render RRTCircularProgress component with color=secondary prop 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<RRTCircularProgress /> should render RRTCircularProgress component with keepHeight prop 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root"
    style="height: 420px;"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
